src/chart/context.rs

```diff
@@ -549,57 +516,54 @@ impl<'a, DB: DrawingBackend, X: Ranged, Y: Ranged> ChartContext<'a, DB, RangedCo
                 continue;
             }
 
-            /* Then we need to estimate the text if rendered */
-            let (w, h) = self
-                .drawing_area
-                .estimate_text_size(&t, &label_style.font)
-                .unwrap_or((0, 0));
-
-            let (cx, cy) = if tick_size >= 0 {
+            let (cx, cy, h_pos, v_pos) = if tick_size >= 0 {
                 match orientation {
                     // Right
                     (dx, dy) if dx > 0 && dy == 0 => {
-                        (right_alignment + label_dist - w as i32, *p - y0)
+                        (label_dist, *p - y0, HPos::Left, VPos::Center)
```

Comment:
Could we use right alignment here?

Reply:
The labels on the right side of the chart use the left alignment as before, nothing changed here. The name `right_alignment` was quite confusing as it was used in a meaning "the max width of the labels" (therefore the calculations produced the proper offset for all labels being on the same vertical line with the distance of `label_dist` from the axis).

Reply:
I believe there's a change here. Since before the change, we just make sure the right aligned for all the labels. 
Sorry about the confusing, but this is what I mean.

Before the change:
![image](https://user-images.githubusercontent.com/3023614/69563115-b625bd80-0f6d-11ea-96db-0c2ca0948483.png)

After the change:
![image](https://user-images.githubusercontent.com/3023614/69563160-ca69ba80-0f6d-11ea-92d0-efffc78cb60f.png)

I still think getting the max width is some step we should do here

Reply:
Oh, I see, actually I was thinking it's a bug and never realized that it's intended :) Please let me explain - usually on the charts the right labels are aligned by the left side as the width could differ a lot. For your use-case the right alignment looks better because the digits are aligned (and for usual linear scales that probably would be better). However, for logarithmic and category labels it could be not so nice (for example, legitimate IPv4/IPv6 addresses):

![boxplot](https://user-images.githubusercontent.com/382655/69571606-a7e09d00-0f7f-11ea-9f3a-1ea4a87ab993.png)

The mapping between the ticks and labels is not clearly visible. I could suggest 2 options:
1. Add an option for the alignment (now sure how the API would look like though);
2. Align based on the label source (category vs numeric etc, but again not sure is it even possible).

Unfortunately, I need the left alignment (as labels on my boxplot chart have quite different widths) and would prefer to avoid any hacks and find the proper solution.